modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -204,6 +216,21 @@ private int compareWatermarks(Serializable w1, Serializable w2, Comparator compa
     return comparator.compare(w1, w2);
   }
 
+  @Override
+  public RestartContext getRestartContext() {
+    restarting.set(true);
+    delegateRunnable.setDelegate(null);
+    return new RestartContext(executor, delegateRunnable);
+  }
+
+  @Override
+  public void restart(RestartContext restartContext) {
+    restarting.set(true);
+
+    executor = restartContext.getExecutor();
+    delegateRunnable = restartContext.getDelegateRunnable();
```

Comment:
this is still wrong. If you do this here.. the poll() method that this delegates into is the poll from the old instance you just discarded. this should be
```
delegateRunnable = restartContext.getDelegateRunnable();
delegateRunnable.setDelegate(() -> poll());
```

Reply:
At this point the delegate is null, I can set it to null here instead of in the getRestartContext method, but I cannot set the delegate here since the poll method needs the SourceCallbackContext which is not available yet, we get it in the doStart method.

Reply:
then it is clear that this should happen in the doStart() method instead of here.........